### ISSUE TITLE
United Healthcare broken

### ISSUE THREAD

**Issue by AnalogJ** [fastenhealth/fasten-onprem#318]
After users authenticate via United Healthcare, they get the following error message:

![image(1)](https://github.com/fastenhealth/fasten-onprem/assets/891875/aea0cad4-9db8-4db7-af08-9543bbcd1971)

At which point they are redirected to the Fasten app. 
![err](https://github.com/fastenhealth/fasten-onprem/assets/891875/4031ad4c-85ab-47e6-8704-a5d7f4ef4198)

---

A similar error (but better user experience) occurs when authenticating with Flexpa

![image](https://github.com/fastenhealth/fasten-onprem/assets/891875/168fed33-7405-4c09-8b11-226be16638ec)

---

After contacting UHC vendor support, they responded with the following message (emphasis mine):

> FLEX was designed to be in compliance with the[ CMS Final rule](https://www.federalregister.gov/documents/2020/05/01/2020-05050/medicare-and-medicaid-programs-patient-protection-and-affordable-care-act-interoperability-and). Eligibility errors generally occurs when the member has a UHC plan but not one that falls under the umbrella of that rule so they are not currently eligible for FLEX. If a member sees an eligibility error please confirm **they have a plan that falls into one of the categories listed there and not a commercial plan.**

This is similar to the error that Aetna users encounter - #33

**Comment by ebdavison** [comment]
I just tried UHC and got the same red error message.  But, I did not get the other.  My experience was that I was able to login properly to UHC and get redirected back to fasten.  No other error is displayed on my end.

However, in the docker logs I get several errors like this:

```
time="2023-11-16T23:34:15Z" level=info msg="10.211.2.53 - eef8eb92db63 [16/Nov/2023:23:34:15 +0000] \"GET /web/sources/callback/assets/img/apple-icon.png\" 200 9159 \"http://10.211.1.114:9090/web/sources/callback/unitedhealthcare?error=server_error&state=155edc38-affe-487e-8a94-df2fbdb633a0&error_id=1260631435004542-%5B5%2C+4%2C+b%2C+b%2C+8%2C+0%2C+b%2C+a%2C+5%2C+4%5D&error_description=The+eligibility+service+failed.\" \"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/119.0\" (1ms)" clientIP=10.211.2.53 hostname=eef8eb92db63 latency=1 method=GET path=/web/sources/callback/assets/img/apple-icon.png referer="http://10.211.1.114:9090/web/sources/callback/unitedhealthcare?error=server_error&state=155edc38-affe-487e-8a94-df2fbdb633a0&error_id=1260631435004542-%5B5%2C+4%2C+b%2C+b%2C+8%2C+0%2C+b%2C+a%2C+5%2C+4%5D&error_description=The+eligibility+service+failed." respLength=9159 statusCode=200 type=web userAgent="Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/119.0"
```

I can send all log lines if needed.

**Comment by chrishas35** [comment]
> After contacting UHC vendor support, they responded with the following message (emphasis mine):
> 
> > FLEX was designed to be in compliance with the[ CMS Final rule](https://www.federalregister.gov/documents/2020/05/01/2020-05050/medicare-and-medicaid-programs-patient-protection-and-affordable-care-act-interoperability-and). Eligibility errors generally occurs when the member has a UHC plan but not one that falls under the umbrella of that rule so they are not currently eligible for FLEX. If a member sees an eligibility error please confirm **they have a plan that falls into one of the categories listed there and not a commercial plan.**

A provider not supporting commercial plans feels like a pretty big caveat that should be called out in documentation.

**Comment by AnalogJ** [comment]
@chrishas35 Agreed. This is definitely not ideal, and is something we're discussing on Discord. 

Thankfully this only impacts Claims & Benefits records -- not Health/Medical records. So you should be able to find your hospital/hospital network in Fasten and pull your records directly from them.

**Comment by geoffrothman** [comment]
> > After contacting UHC vendor support, they responded with the following message (emphasis mine):
> > > FLEX was designed to be in compliance with the[ CMS Final rule](https://www.federalregister.gov/documents/2020/05/01/2020-05050/medicare-and-medicaid-programs-patient-protection-and-affordable-care-act-interoperability-and). Eligibility errors generally occurs when the member has a UHC plan but not one that falls under the umbrella of that rule so they are not currently eligible for FLEX. If a member sees an eligibility error please confirm **they have a plan that falls into one of the categories listed there and not a commercial plan.**
> 
> A provider not supporting commercial plans feels like a pretty big caveat that should be called out in documentation.

@AnalogJ I'm facing the same issue with UHC. Can you put something under hte UHC icon that gives them a heads up that this might happen...i just spent a few hours trying to troubleshoot it and it almost pushed me over the edge of keeping on with the project because of that experience.
